rfcs/InputUnion.md

```diff
@@ -133,11 +133,31 @@ In this mutation, we encounter the main challenge of the **Input Union** - we ne
 
 A wide variety of solutions have been explored by the community, and they are outlined in detail in this document under [Possible Solutions](#Possible-Solutions).
 
-## Requirements
+## Solution Shape - Aims & Objections
 
-1. **Doesn't inhibit [schema evolution](https://graphql.github.io/graphql-spec/draft/#sec-Validation.Type-system-evolution)** ─ adding a new member type to an Input Union or doing any non-breaking change to existing member types must not result in breaking change. For example, adding a new optional field to member type or changing a field from non-nullable to nullable must not break previously valid client operations.
+Hypothetical goals that a solution might attempt to fulfill.
 
-**Note**: Input Unions may enforce some restrictions on member types, but if they do so compliance with these restrictions must be fully validated during schema building (analagous to how interfaces enforce restrictions on member types).
+### GraphQL should contain an Input Union type
```

Comment:
I think that `Input Union` is a charged term and actually shapes how people think the solution should be shaped. Personally I prefer the more neutral "polymorphic input" term, since this is actually what we're after, and it means that we do not have to mirror the output types opening more potential solutions (with potentially better trade-offs).

```suggestion
### GraphQL should contain a polymorphic Input type
```